ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/submission/FibonacciCommandGenerator.scala

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.benchtool.submission
+
+import com.daml.ledger.api.benchtool.config.WorkflowConfig.FibonacciSubmissionConfig
+import com.daml.ledger.api.v1.commands.{Command, CreateAndExerciseCommand}
+import com.daml.ledger.api.v1.value.{Identifier, Record, RecordField, Value}
+import com.daml.ledger.client.binding.Primitive
+import com.daml.ledger.test.model.Bench.InefficientFibonacci.toNamedArguments
+
+import scala.util.{Success, Try}
+
+final class FibonacciCommandGenerator(
+    config: FibonacciSubmissionConfig,
+    signatory: Primitive.Party,
+) extends CommandGenerator {
+
+  def next(): Try[Seq[Command]] = {
+    Success(
+      Seq(
+        Command(
+          Command.Command.CreateAndExercise(
+            CreateAndExerciseCommand(
+              templateId = Some(
+                com.daml.ledger.test.model.Bench.InefficientFibonacci.id.asInstanceOf[Identifier]
+              ),
+              createArguments = Some(
+                toNamedArguments(com.daml.ledger.test.model.Bench.InefficientFibonacci(signatory))
+              ),
+              choice = "InefficientFibonacci_Compute",
+              choiceArgument = Some(
+                Value(
+                  Value.Sum.Record(
+                    Record(
+                      None,
+                      Seq(
+                        RecordField(
+                          label = "value",
+                          value = Some(Value(Value.Sum.Int64(config.value.toLong))),
+                        )
+                      ),
+                    )
+                  )
+                )
+              ),
+            )
+          )
+        )
+      )
+    )
+  }
```

Comment:
wow, this is huge nesting ;) can it be made more readable by introducing some intermediate vals? :)

Reply:
It is, the wonder of auto-formatting ;-)  Although it is large I don't think it causes any confusion though.